flyway.groovy

```diff
@@ -0,0 +1,77 @@
+#!./lib/runner.groovy
+import com.gargoylesoftware.htmlunit.WebClient
+import com.gargoylesoftware.htmlunit.html.HtmlAnchor
+import com.gargoylesoftware.htmlunit.html.HtmlPage
+import hudson.util.VersionNumber
+import net.sf.json.JSONObject
+
+import java.util.regex.Pattern
+// Generates server-side metadata for Flyway command-line
+def webclient (){
+  def wc = new WebClient()
+  wc.javaScriptEnabled = false;
+  wc.cssEnabled = false;
+  return wc
+}
+
+def listFromMavenRepo() {
+  def json = []
+  def url = "https://repo1.maven.org/maven2/org/flywaydb/flyway-commandline/";
+
+  def wc=webclient();
+  HtmlPage p = wc.getPage(url);
+  def pattern = Pattern.compile("^([0-9][0-9\\.]+)/\$");
+
+  p.getAnchors().collect { HtmlAnchor a ->
+    m = pattern.matcher(a.hrefAttribute)
+    if (m.find()) {
+      ver = m.group(1)
+      json.addAll( os_versions(url + ver, ver, wc))
+    }
+  }
+  return json
+}
+
+def os_versions(url, version, wc) {
+  json=[]
+  HtmlPage page= wc.getPage(url)
+  def pattern = Pattern.compile("flyway-commandline-" + ver + "(\\.tar\\.gz|-([a-z]+)-x64(\\.zip|\\.tar.gz))\$")
```

Comment:
It is indeed restrictive to those packages that the accompanying installer knows how to handle.  The pattern is an attempt to gather the 4 available platform assemblies of each flyway version (and make it easy to extract each of those assemblies's name, i.e. windows, linux, etc).

I'm open to suggestions on how to do this better.  For example, I could parse each version's pom to divine what assemblies exist.